inc/composer/class-command.php

```diff
@@ -654,6 +661,179 @@ protected function db( InputInterface $input, OutputInterface $output ) {
 		return $return_val;
 	}
 
+	/**
+	 * Generate SSL certificates for development environment.
+	 *
+	 * @param InputInterface $input Command input object.
+	 * @param OutputInterface $output Command output object.
+	 * @return int
+	 */
+	protected function ssl( InputInterface $input, OutputInterface $output ) {
+		$subcommand = $input->getArgument( 'options' )[0] ?? null;
+
+		switch ( $subcommand ) {
+			case 'install':
+				// Detect platform architecture to attempt automatic installation.
+				$os = php_uname( 's' ); # 'Darwin', 'Linux', 'Windows'
+				$arch = php_uname( 'm' ); # 'arm64' for arm, 'x86_64' or 'amd64' for x64
+				$mkcert_version = 'v1.4.3';
+
+				switch( $os ) {
+					# macOS
+					case 'Darwin':
+						$binary_arch = $arch === 'x86_64' ? 'darwin-amd64' : 'darwin-arm64';
+						break;
+					# Linux
+					case 'Linux':
+						$binary_arch = $arch === 'amd64' ? 'linux-amd64' : 'linux-arm64';
+						break;
+					# Windows
+					case 'Windows':
+						$binary_arch = 'windows-amd64.exe';
```

Comment:
Will need to check what the situation with this and WSL is. If you can run .exe from WSL, or global commands from there we might be ok. @missjwo might be able to assist in testing.

Reply:
Yes, testing cross-platform was supposed to be the next step. I can do that in a VM.

Reply:
WSL won’t replicate accurately within a VM so far as I know, since it’s a VM itself.

For WSL though, the arch will be Linux because it’s literally a Linux VM you’re running inside of.

Reply:
If composer is running within WSL, the certificate would be automatically installed to Windows store itself, which is a problem. We'll need to add docs on manually accepting the root CA.

Also, does this mean we do not support Windows itself but rather WSL ? ( I have not used WSL nor Docker on Windows )

Reply:
Yeah there was a PR to support windows natively but I think there were some limitations to that. We do have a way to check it’s a WSL environment at least.

Reply:
You can run .exe files from WSL https://docs.microsoft.com/en-us/windows/wsl/filesystems

Reply:
Happy to test stuff - just let me know when the PR is ready to be tested and what i should be looking out for.

Reply:
@shadyvb Correct, only WSL is supported, per https://docs.altis-dxp.com/local-server/windows/

You can execute exe files, yeah; WSL will transparently map execution of those, and execute them. Note though, `php_uname( 's' )` will return Linux because it is Linux. We need to use the `Command::is_wsl()` function to detect WSL, not uname.

Also happy to test with my WSL env, and screenshare as needed.